### DISCUSSION TITLE
Decide Languages for the monorepo

### DISCUSSION THREAD

**Discussion by a-man-called-q** [rivanadotdev/waha#8] [Ideas]
For the `MVP` we might use `Typescript` with `allowJS: true`. This way we can speed up a bit on prototyping and achieving what we want.

However, there is a slight change that moving straight away to `Rust` might be possible. **Only if** the diagrams are well received and easy to read.

**Comment by a-man-called-q** [comment]
Oh as for now, we're using moonrepo so it supports `nodejs` and partial support for `rust` out of the box. @yosephfernando any opinion ?

**Reply by yosephfernando** [reply]
I think we need to create the lint first to make sure the folder structure and syntax consistency

**Reply by a-man-called-q** [reply]
Indeed, we should create the lint rules. However, we have to decide which language we want to use. So are we going for `NodeJS` or straight away doing `Rust` since we have no deadline at this point.

**Reply by yosephfernando** [reply]
lets use rust but only for 1 service, after that we evaluate the service. If its worth it than we can implement all service using rust. i suggest reward service to start with. what do you think ?

**Reply by a-man-called-q** [reply]
Agreed. but I don't think we should use `NodeJS` in one and `Rust` in others. 

@yosephfernando wrote:
> after that we evaluate the service. If its worth it than we can implement all service using rust.

I think we can use `NodeJS` for all the services for now. Let's focus on our NodeJS code structure then.

**Reply by a-man-called-q** [reply]
This will be closed in favor of #10

**Comment by a-man-called-q** [comment]
However reward-service is highly dependant on quest-service.I will check your code tomorrow and I will give my opinion on this.On Nov 16, 2024 10:40 PM, yosephfernando ***@***.***> wrote:
lets use rust but only for 1 service, after that we evaluate the service. If its worth it than we can implement all service using rust. i suggest reward service to start with. what do you think ?
